audio/out/buffer.c

```diff
@@ -319,6 +401,32 @@ double ao_get_delay(struct ao *ao)
     return driver_delay + pending / (double)ao->samplerate;
 }
 
+double ao_get_playing_pts(struct ao *ao)
+{
+    struct buffer_state *p = ao->buffer_state;
+
+    mp_mutex_lock(&p->lock);
+
+    int pos;
+    double cur_time = MP_TIME_NS_TO_S(mp_time_ns());
+
+    for (pos = 0; pos <= p->num_queued_aframes; pos++) {
+        if (pos == p->num_queued_aframes || p->queued_aframes[pos].start_time > cur_time)
+            break;
+    }
```

Comment:
```suggestion
    for (pos = 0; pos < p->num_queued_aframes; pos++) {
        if (p->queued_aframes[pos].start_time > cur_time)
            break;
    }
```
should be the same, no? Also, shouldn't that bet `>= cur_time` check?

Reply:
I have mentioned it in https://github.com/mpv-player/mpv/pull/14058#discussion_r1874027261. I haven't pushed my local modifications yet. Sorry.